benchmarks/run_bayesmark.py

```diff
@@ -0,0 +1,159 @@
+import argparse
+import json
+import os
+import subprocess
+from typing import Dict
+from typing import List
+
+from matplotlib import cm
+from matplotlib import colors
+from matplotlib.axes import Axes
+import matplotlib.pyplot as plt
+import numpy as np
+from xarray import Dataset
+
+import bayesmark.constants as cc
+from bayesmark.serialize import XRSerializer
+
+
+_DB = "bo_optuna_run"
+
+
+def run_benchmark(args: argparse.Namespace) -> None:
+
+    sampler_list = args.sampler_list.split()
+    sampler_kwargs_list = args.sampler_kwargs_list.split()
+    pruner_list = args.pruner_list.split()
+    pruner_kwargs_list = args.pruner_kwargs_list.split()
+
+    config = dict()
+    for sampler, sampler_kwargs in zip(sampler_list, sampler_kwargs_list):
+        for pruner, pruner_kwargs in zip(pruner_list, pruner_kwargs_list):
+            optimizer_name = f"{sampler}-{pruner}-Optuna"
+            optimizer_kwargs = {
+                "sampler": sampler,
+                "sampler_kwargs": json.loads(sampler_kwargs),
+                "pruner": pruner,
+                "pruner_kwargs": json.loads(pruner_kwargs),
+            }
+            # We need to dynamically generate config.json sice sampler pruner combos (solvers)
+            # are parametrized by user. Following sample config schema.
+            # https://github.com/uber/bayesmark/blob/master/example_opt_root/config.json
+            config[optimizer_name] = ["optuna_optimizer.py", optimizer_kwargs]
+
+    with open(os.path.join("benchmarks", "bayesmark", "config.json"), "w") as file:
+        json.dump(config, file, indent=4)
+
+    samplers = " ".join(config.keys())
+    metric = "nll" if args.dataset in ["breast", "iris", "wine", "digits"] else "mse"
+    cmd = (
+        f"bayesmark-launch -n {args.budget} -r {args.repeat} "
+        f"-dir runs -b {_DB} "
+        f"-o RandomSearch {samplers} "
```

Comment:
Note - Builtin random sampler is used here since it's automatically detected by bayesmark to be used as baseline for leaderboards. Otherwise we would have to specify one.

Reply:
This way `bayesmark` automatically derives baseline run. From the doc
> The bayesmark-anal command looks for a baseline.json file in [DB_ROOT]/[DBID]/derived, which states the best possible and random search performance. If no such file is present, bayesmark-anal automatically calls bayesmark-baseline to build it. The baselines are inferred from the random search performance in the logs. The baseline values are considered fixed (not random) quantities when bayesmark-anal builds confidence intervals. Therefore, we allow the user to leave them fixed and do not rebuild them when bayesmark-anal is called if a baselines file is already present.

Without baseline run benchmark [will fail](https://github.com/xadrianzetx/dev-optuna-bayesmark/runs/5566869587?check_suite_focus=true). We could use Optuna random or grid samplers instead, but that's more code for essentially the same effect.